src/components/TextField/TextField.vue

```diff
@@ -0,0 +1,253 @@
+	<div class="wrapper">
+		<TextField :value.sync="text1"
+			label="Leading icon and trailing button"
+			trailing-button-icon="close"
+			:show-trailing-button="text1 !== ''"
+			@trailing-button-click="clearText">
+			<Magnify :size="16" />
+		</TextField>
+		<TextField :value.sync="text2"
+			label="Success state"
+			:success="true"
+			@trailing-button-click="clearText">
+		</TextField>
+		<TextField :value.sync="text3"
+			label="Error state"
+			:error="true"
+			@trailing-button-click="clearText">
+		</TextField>
+		<TextField :value.sync="text4"
+			label="Internal label"
+			placeholder="That can be used together with placeholder"
+			:label-visible="true"
+			trailing-button-icon="close"
+			:show-trailing-button="text4 !== ''"
+			@trailing-button-click="clearText">
+			<Lock :size="16" />
+		</TextField>
+		<div class="external-label">
+			<label for="$refs.textField.id">External label</label>
+			<TextField :value.sync="text5"
+				ref="textField"
+				:label-outside= "true"
+				@trailing-button-click="clearText" />
+		</div>
+	</div>
+</template>
+<script>
+import Magnify from 'vue-material-design-icons/Magnify'
+import Lock from 'vue-material-design-icons/Lock'
+import Close from 'vue-material-design-icons/Close'
+
+export default {
+	data() {
+		return {
+			text1: '',
+			text2: '',
+			text3: '',
+			text4: '',
+			text5: '',
+		}
+	},
+
+	components: {
+		Magnify,
+		Lock,
+		Close,
+	},
+
+	methods: {
+		clearText() {
+			this.text1 = ''
+			this.text3 = ''
+		}
+	}
+}
+</script>
+<style lang="scss" scoped>
+.wrapper {
+	display: flex;
+	gap: 4px;
+	align-items: flex-end;
+	flex-wrap: wrap;
+}
+
+.external-label {
+	display: flex;
+	width: 100%;
+}
+</style>
+```
+</docs>
+
+<template>
+	<InputField v-bind="$props"
+		v-on="$listeners"
+		@input="handleInput">
+		<!-- Default slot for the leading icon -->
+		<slot />
+		<template slot="trailing-button-icon">
+			<Close v-if="trailingButtonIcon === 'close'" :size="20" />
+			<ArrowRight v-else-if="trailingButtonIcon === 'arrowRight'" :size="20" />
+		</template>
+	</InputField>
+</template>
+
+<script>
+
+import InputField from '../InputField/InputField.vue'
+import Close from 'vue-material-design-icons/Close'
+import ArrowRight from 'vue-material-design-icons/ArrowRight'
+
+export default {
+	name: 'TextField',
+
+	components: {
+		InputField,
+		Close,
+		ArrowRight,
+	},
+
+	props: {
+		/**
+		 * The value of the input field
+		 */
+		value: {
+			type: String,
+			required: true,
+		},
+
+		/**
+		 * The type of the input element, it can be `text`, `password`,
+		 * `email`, `number`, `tel` and `url`.
+		 */
+		type: {
+			type: String,
+			default: 'text',
+			validator: (value) => [
+				'text',
+				'password',
+				'email',
+				'tel',
+				'url',
+			].includes(value),
+		},
+
+		/**
+		 * The hidden input label for accessibility purposes. This will also
+		 * be used as a placeholder unless the placeholder prop is populated
+		 * with a different string.
```

Comment:
**Strongly suggesting** Should state that it's required when labelOutside is not set.